api/insight/socket.io.go

```diff
@@ -145,68 +158,104 @@ func (soc *SocketServer) Store(blockData *blockdata.BlockData, msgBlock *wire.Ms
 	// Since the coinbase transaction is generated by the miner, it will never
 	// hit mempool. It must be processed now, with the new block.
 	coinbaseTx := msgBlock.Transactions[0]
-	coinbaseHash := coinbaseTx.TxHash().String()
-	// Check each output's pkScript for subscribed addresses.
-	for _, out := range coinbaseTx.TxOut {
-		_, scriptAddrs, _, err := txscript.ExtractPkScriptAddrs(
-			out.Version, out.PkScript, soc.params)
-		if err != nil {
-			apiLog.Warnf("failed to decode pkScript: %v", err)
-			continue
-		}
-
-		if len(scriptAddrs) == 0 {
-			continue
-		}
-
-		soc.watchedAddresses.RLock()
-		for _, address := range scriptAddrs {
-			addr := address.String()
-			if _, ok := soc.watchedAddresses.c[addr]; ok {
-				apiLog.Debugf(`Broadcasting "%s" in "%s" to subscribed clients.`,
-					addr, coinbaseHash)
-				soc.BroadcastTo(addr, addr, coinbaseHash)
-			}
-		}
-		soc.watchedAddresses.RUnlock()
+	coinbaseHex, err := txhelpers.MsgTxToHex(coinbaseTx)
+	if err != nil {
+		apiLog.Errorf("Failed to decode tx %v: %v", coinbaseTx, err)
+		return err
+	}
 
+	soc.newTxChan <- &NewTx{
+		Hex: coinbaseHex,
```

Comment:
Subscribed address checking for the coinbase transaction is now handled in `relayNewTxs` with all the other transactions.